topics/ecology/tutorials/x-array-map-plot/tutorial.md

```diff
@@ -0,0 +1,460 @@
+>
+> 1. What are the different kinds of colormaps that can be used?
+>
+> > ### {% icon solution %} Solution
+> >
+> >When it comes to conveying the correct information, through visualisation, colors play a major role. Suppose you want to display a cold region, its an obvious practice of using cooler tones such as a `blue`. Thus it is important to understand the choices we have. Find the [complete documentation of ColorMaps available here](https://github.com/Quickbeasts51429/Xarray_ColorMaps/blob/main/index.md)
+> >
+> >
+> >## Plotting different colormaps using **NetCDF xarray map plotting**
+ > > > ### {% icon hands_on %} Hands-on: Plotting the major colormaps using the xarray tool.
+> > >
+> > > 1. {% tool [NetCDF xarray map plotting](toolshed.g2.bx.psu.edu/repos/ecology/xarray_mapplot/xarray_mapplot/0.20.2+galaxy0) %} with the following parameters:
+> > >    - {% icon param-file %}
+*"Netcdf file"*: `outfile_18.nc`
+> > >    - {% icon param-file %} *"Tabular of variables"*: `Metadata infos from outfile_18.nc` (output of **NetCDF xarray Metadata Info** {% icon tool %})
+> > >    - *"Choose the variable to plot"*: `air_temperature_at_2_metres`
+> > >    - *"Name of latitude coordinate"*: `lat`
+> > >    - *"Name of longitude coordinate"*: `lon`
+> > >    - *"Datetime selection"*: `No`
+> > >    - *"Range of values for plotting e.g. minimum value and maximum value (minval,maxval) (optional)"*: `220,320`
+> > >    - *"Add country borders with alpha value [0-1] (optional)"*: `1.0`
+> > >    - *"Add coastline with alpha value [0-1] (optional)"*: `1.0`
+> > >    - *"Add ocean with alpha value [0-1] (optional)"*: `1.0`
+> > >    - *"Specify plot title (optional)"*: `Projection :  Mercator 18:00 UTC `
+> > >    - *"Specify which colormap to use for plotting (optional)"*: `CMRmap`
+> > >    ![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 at 18:00:00 : CMRmap](../../images/colorselect.png)
+> > >    - *"Specify the projection (proj4) on which we draw e.g. {"proj":"PlateCarree"} with double quote (optional)"*: `{'proj': 'Mercator'}`
+> > >
+> > >
+> > > The final plot is shown below: 
+> > >    ![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 >at 18:00:00](../../images/CMRmap.png)
+> > >Some other important color variants of the same map can be found below :
+> > > - *colormap : Accent*
+> > >![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 >at 18:00:00](../../images/Accent.png)
+> > >  - *colormap : bam*
+> > >![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 >at 18:00:00](../../images/bam.png)
+> > >  - *colormap : bukavu*
+> > >![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 >at 18:00:00](../../images/bukavu.png)
+> > > - *colormap : coolwarm*
+> > >![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 >at 18:00:00](../../images/coolwarm.png)
+> > > - *colormap : jet*
+> > >![ECMWF Reanalysis Air temperature a 2 metres on 2022-05-25 >at 18:00:00](../../images/jet.png)
```

Comment:
As far as I know jet is not a colorblind friendly colormap. For Earth Science, we usually recommend to use https://www.fabiocrameri.ch/colourmaps/ which I think are available in the xarray map tool.